ts-obs/cmd/kubectl.go

```diff
@@ -1,302 +1,332 @@
 package cmd
 
 import (
-    "context"
-    "fmt"
-    "io"
-    "log"
-    "net/http"
-    "os"
-
-    corev1 "k8s.io/api/core/v1"
-    "k8s.io/client-go/kubernetes"
-    "k8s.io/client-go/kubernetes/scheme"
-    rest "k8s.io/client-go/rest"
-    "k8s.io/client-go/transport/spdy"
-    "k8s.io/client-go/tools/clientcmd"
-    "k8s.io/client-go/tools/portforward"
-    "k8s.io/client-go/tools/remotecommand"
-    metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-    "k8s.io/apimachinery/pkg/labels"
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	rest "k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/portforward"
+	"k8s.io/client-go/tools/remotecommand"
+	"k8s.io/client-go/transport/spdy"
 )
 
-func kubeInit() (kubernetes.Interface, *rest.Config) {
-    var err error
+func KubeInit() (kubernetes.Interface, *rest.Config) {
+	var err error
 
-    config, err := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
-		&clientcmd.ClientConfigLoadingRules{ExplicitPath: os.Getenv("HOME") + "/.kube/config"},
-		&clientcmd.ConfigOverrides{CurrentContext: "minikube"}).ClientConfig()
-    if err != nil {
-        log.Fatal(err)
-    }
+	config, err := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
+		&clientcmd.ClientConfigLoadingRules{ExplicitPath: os.ExpandEnv("$HOME/.kube/config")},
```

Comment:
maybe we should put $HOME in a var so we can override it when building for Windows? (we used this for timescale-prometheus https://www.digitalocean.com/community/tutorials/using-ldflags-to-set-version-information-for-go-applications)